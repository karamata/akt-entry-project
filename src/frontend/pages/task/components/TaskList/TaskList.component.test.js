import React from 'react';
import renderer from 'react-test-renderer';

import TableTaskList from './TaskList.component';

jest.mock('antd/dist/antd.css', () => <div />);

it('Create task list component no item', () => {
  const component = renderer.create(<TableTaskList />);
  expect(component).toMatchSnapshot();
});

it('Create task list component has tasks', () => {
  const component = renderer.create(<TableTaskList tasks={[{ _id: '10000', title: 'Do homework' }]} />);
  expect(component).toMatchSnapshot();
});
