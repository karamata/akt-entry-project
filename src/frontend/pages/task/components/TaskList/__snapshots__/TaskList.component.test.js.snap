// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Create task list component has tasks 1`] = `
<div
  className="ant-table-wrapper"
>
  <div
    className="ant-spin-nested-loading"
  >
    <div
      className="ant-spin-container"
    >
      <div
        className="ant-table ant-table-default ant-table-without-column-header ant-table-scroll-position-left"
        style={Object {}}
      >
        <div
          className="ant-table-content"
        >
          <div
            className="ant-table-body"
            onScroll={[Function]}
            onWheel={[Function]}
            style={Object {}}
          >
            <table
              className=""
              style={Object {}}
            >
              <colgroup>
                <col
                  style={
                    Object {
                      "minWidth": "80%",
                      "width": "80%",
                    }
                  }
                />
                <col
                  style={
                    Object {
                      "minWidth": undefined,
                      "width": undefined,
                    }
                  }
                />
              </colgroup>
              <tbody
                className="ant-table-tbody"
              >
                <tr
                  className="ant-table-row ant-table-row-level-0"
                  data-row-key="10000"
                  onClick={[Function]}
                  onContextMenu={[Function]}
                  onDoubleClick={[Function]}
                  onMouseEnter={[Function]}
                  onMouseLeave={[Function]}
                  style={
                    Object {
                      "height": null,
                    }
                  }
                >
                  <td
                    className="ant-table-row-cell-break-word"
                    onClick={[Function]}
                  >
                    <div>
                      <label
                        className="ant-checkbox-wrapper"
                      >
                        <span
                          className="ant-checkbox"
                          style={Object {}}
                        >
                          <input
                            checked={false}
                            className="ant-checkbox-input"
                            onBlur={[Function]}
                            onChange={[Function]}
                            onFocus={[Function]}
                            type="checkbox"
                          />
                          <span
                            className="ant-checkbox-inner"
                          />
                        </span>
                      </label>
                       
                      Do homework
                    </div>
                  </td>
                  <td
                    className=""
                    onClick={[Function]}
                  >
                    <button
                      className="ant-btn ant-btn-danger"
                      onClick={[Function]}
                      onMouseDown={[Function]}
                      onTouchStart={[Function]}
                      style={
                        Object {
                          "marginRight": 5,
                        }
                      }
                      type="button"
                    >
                      <span>
                        Delete
                      </span>
                    </button>
                    <button
                      className="ant-btn"
                      onClick={[Function]}
                      style={
                        Object {
                          "backgroundColor": "#28A745",
                          "color": "#fff",
                        }
                      }
                      type="button"
                    >
                      <span>
                        Edit
                      </span>
                    </button>
                  </td>
                </tr>
              </tbody>
            </table>
          </div>
        </div>
      </div>
    </div>
  </div>
</div>
`;

exports[`Create task list component no item 1`] = `
<div
  className="ant-table-wrapper"
>
  <div
    className="ant-spin-nested-loading"
  >
    <div
      className="ant-spin-container"
    >
      <div
        className="ant-table ant-table-default ant-table-empty ant-table-without-column-header ant-table-scroll-position-left"
        style={Object {}}
      >
        <div
          className="ant-table-content"
        >
          <div
            className="ant-table-body"
            onScroll={[Function]}
            onWheel={[Function]}
            style={Object {}}
          >
            <table
              className=""
              style={Object {}}
            >
              <colgroup>
                <col
                  style={
                    Object {
                      "minWidth": "80%",
                      "width": "80%",
                    }
                  }
                />
                <col
                  style={
                    Object {
                      "minWidth": undefined,
                      "width": undefined,
                    }
                  }
                />
              </colgroup>
              <tbody
                className="ant-table-tbody"
              />
            </table>
          </div>
          <div
            className="ant-table-placeholder"
          >
            <div
              className="ant-empty ant-empty-normal"
            >
              <div
                className="ant-empty-image"
              >
                <svg
                  height="41"
                  viewBox="0 0 64 41"
                  width="64"
                  xmlns="http://www.w3.org/2000/svg"
                >
                  <g
                    fill="none"
                    fillRule="evenodd"
                    transform="translate(0 1)"
                  >
                    <ellipse
                      cx="32"
                      cy="33"
                      fill="#F5F5F5"
                      rx="32"
                      ry="7"
                    />
                    <g
                      fillRule="nonzero"
                      stroke="#D9D9D9"
                    >
                      <path
                        d="M55 12.76L44.854 1.258C44.367.474 43.656 0 42.907 0H21.093c-.749 0-1.46.474-1.947 1.257L9 12.761V22h46v-9.24z"
                      />
                      <path
                        d="M41.613 15.931c0-1.605.994-2.93 2.227-2.931H55v18.137C55 33.26 53.68 35 52.05 35h-40.1C10.32 35 9 33.259 9 31.137V13h11.16c1.233 0 2.227 1.323 2.227 2.928v.022c0 1.605 1.005 2.901 2.237 2.901h14.752c1.232 0 2.237-1.308 2.237-2.913v-.007z"
                        fill="#FAFAFA"
                      />
                    </g>
                  </g>
                </svg>
              </div>
              <p
                className="ant-empty-description"
              >
                No Data
              </p>
            </div>
          </div>
        </div>
      </div>
    </div>
  </div>
</div>
`;
